### ISSUE TITLE
Timeouts for view requests when syncing can leave changes requests hanging forever

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7759]
**Describe the bug**
When an offline user requests `_changes`, two view requests are required to determine which documents the user has access to. 
If one of the view requests times out because the view is not indexed, the initiating `_changes` request hangs on forever. 

**To Reproduce**
1. Start with a hefty database, indexing the `medic` ddoc view should take over 5 minutes. 
2. Login as an offline user. 
3. Edit `docs_by_replication_key` view in the `medic` ddoc. The change should be inconsequential, even editing or adding a comment.
4. Start indexing. 
5. Sync with your offline user. 
6. Watching the api logs, you should see something like: 
```
2022-09-01 07:54:58 ERROR: Server error: {
  error: 'timeout',
  reason: 'The request could not be processed in a reasonable amount of time.',
  status: 500,
  name: 'timeout',
  message: 'The request could not be processed in a reasonable amount of time.'
} 
```
This is where your request should have been interrupted. However your client will still show that sync-ing is ongoing.

**Expected behavior**
You should get an error that syncing was not possible. Or, at least, the request should not hang forever. 

**Environment**
- App: api
- Version: 3.10+

**Additional information:**
This is a tricky situation, because of our heartbeat. 
We start sending a heartbeat immediately after we receive the request (before we start setting up required view requests). Sending a heartbeat through means that we've already sent headers and a 200 status. So if the error occurs after the first heartbeat, we can no longer properly communicate the error to the client, through a status. 

Regression after this was fixed in: https://github.com/medic/cht-core/issues/6084

**Comment by dianabarsan** [comment]
I think the easy solution, for now, is to make sure we always end errored requests. 
But from the client's perspective, that 200 means that pouchdb considers that the request succeeded and reports that replication is successful, even though it was not.

**Comment by dianabarsan** [comment]
I think the easy solution, for now, is to make sure we always end errored requests. 
But from the client's perspective, that 200 means that pouchdb considers that the request succeeded and reports that replication is successful, even though it was not.

**Comment by dianabarsan** [comment]
This is ready for AT on `7759-always-end-errored-requests`

**Comment by latin-panda** [comment]
@tatilepizs @dianabarsan just having a look at 3.17 progress.. how is this going? :)

**Comment by ngaruko** [comment]
@latin-panda I am working on this. First attempts to create enough docs to trigger the error have failed. Now trying the [nano](https://github.com/apache/couchdb-nano) library and see how it goes.

**Comment by latin-panda** [comment]
Just want to double check @dianabarsan and @ngaruko 

The 3.17 issues are rebased to master, merged into master and then cherry picked into 3.17.  Yesterday Product (see slack) decided that 3.17 will continue release in 2 weeks.

Do we need this in 3.17?

**Comment by dianabarsan** [comment]
The unable to connect is the correct status and is what we expeted. 

>  invalid json response body 

This indicates that CouhchDb is down and Haproxy responds with html instead of JSON. Also, this changes request is an online user's changes request, and it goes directly to API, so It's unrelated to the issue at hand. 

Can you check API logs at the exact time where you see "The request could not be processed...."? You can even search the logs for that exact error.

**Comment by dianabarsan** [comment]
Any progress on this @ngaruko ?

**Comment by dianabarsan** [comment]
Please let me know if there is anything I can assist with!

**Comment by dianabarsan** [comment]
Merged to `master`
